backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5569920_sys_gh10079ModifyRV_Missing_Counter_Documents.sql

```diff
@@ -0,0 +1,133 @@
+
+
+DROP VIEW IF EXISTS RV_Missing_Counter_Documents;
+
+
+-- This view is made for C_Order entries and their counter-documents
+-- If we start supporting other kind of counter documents in the fututre, this view will have to be updated with the logic on those tables, using unions
+
+CREATE VIEW RV_Missing_Counter_Documents
+AS
+
+
+SELECT
+    -- In the future, if we will add support for other documents (i.e: invoices), we will have to update the id column's logic using a case when logic.
+    -- https://github.com/metasfresh/mf15/issues/318
+    CASE
+        WHEN doc.Record_ID IS NOT NULL THEN (doc.Record_ID * 10) + 1
```

Comment:
```suggestion
        WHEN doc.C_OrderLine_ID IS NOT NULL THEN (doc.Record_ID * 10) + 1
```
i wrote
> explicitly return the C_OrderLine_ID which has the missing counter doc from the doc subselect, together with the Record_ID

exactly for the purpose of using it in `RV_Missing_Counter_Document_ID`

if we later add e.g. C_Invoice_Lines with missing-counter-docs then it will also be aliased to `Record_ID`, so using `doc.Record_ID` does not make sense

Reply:
I thought the invoicelines would be handled in a different left join after looking at the sql and the doc one would only remain for orderlines., but we could also expand the view by joining with the invoice and invoiceline tables on the 'doc' select. Implemented the changes! 😓